### ISSUE TITLE
AzureRmWebAppDeployment: Frequently fails due to file locking issues on target

### ISSUE THREAD

**Issue by oschistad** [Microsoft/vsts-tasks#3498]
Even with the App_Offline flag set, the Azure App Service Deploy task often fails because of a locked file.

While workarounds do exist (use an inactive slot and swap, or use powershell to stop-release-start website), ideally this is something that the deploy task itself ought to handle.

This issue requires some mechanism by which all read locks against the wwwroot are cleared before unpacking the web deploy package.  The simplest solution is to stop the website and start it again post-deploy. We need this in the deploy task itself since we lack a try-catch mechanism inside of the Release workflow.

**Comment by mvvsubbu** [comment]
Sorry for issue you are facing.  This is an issue Azure App Service we have open issue (https://github.com/Microsoft/vsts-tasks/issues/3312) for the same. We will update on that issue once it is fixed. 

For now to unblock your self there is task App Service Management (Preview) which can stop website and start. You can add this task before deployment for stopping and after deployment for starting.

**Comment by mvvsubbu** [comment]
As there is another issue tracking file locking case I am closing this issue.  If the work around doesn't work please feel free to reopen this issue.  Anything related to file locking issue please post it on https://github.com/Microsoft/vsts-tasks/issues/3312
